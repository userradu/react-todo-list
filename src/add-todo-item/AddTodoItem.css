.add-todo-section input {
	margin-right: 5px;
}