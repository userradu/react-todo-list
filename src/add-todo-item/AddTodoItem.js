import React from 'react';
import './AddTodoItem.css';

class AddTodoItem extends React.Component {
	constructor(props) {
		super(props);
		this.state = {itemName: ''};

		this.handleChange = this.handleChange.bind(this);
		this.handleClick = this.handleClick.bind(this);
	}

	handleChange(event) {
		this.setState({itemName: event.target.value});
	}

	handleClick() {
		if (this.state.itemName !== '') {
			this.props.onAddTodoItem(this.state.itemName);
			this.setState({itemName: ''});
		}
	}

	render() {
		return (
			<div className="add-todo-section">
				<input type="text" value={this.state.itemName} onChange={this.handleChange}/>
				<button onClick={this.handleClick}>Add</button>
			</div>
		);
	}
}

export default AddTodoItem;