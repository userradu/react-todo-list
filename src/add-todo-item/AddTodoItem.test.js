import React from 'react';
import AddTodoItem from './AddTodoItem';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';

describe('AddTodoItem', () => {

  it('renders correctly', () => {
    const tree = renderer
      .create(<AddTodoItem onAddTodoItem={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls props.onAddTodoItem()', () => {
    const onAddTodoItem = sinon.spy();
    const wrapper = shallow(<AddTodoItem onAddTodoItem={onAddTodoItem} />);
    const input = wrapper.find('input');
    input.simulate('change', {
      target: { value: 'todoItemName' }
    });
    wrapper.find('button').simulate('click');
    expect(onAddTodoItem.called).toBe(true);
  });

  it('clears the input after calling props.onAddTodoItem()', () => {
    const wrapper = shallow(<AddTodoItem onAddTodoItem={() => {}} />);
    const input = wrapper.find('input');
    input.simulate('change', {
      target: { value: 'todoItemName' }
    });
    wrapper.find('button').simulate('click');
    expect(wrapper.state().itemName).toEqual('');
  });

  it("doesn't call props.onAddTodoItem() when the todo item name is empty", () => {
    const onAddTodoItem = sinon.spy();
    const wrapper = shallow(<AddTodoItem onAddTodoItem={onAddTodoItem} />);
    wrapper.find('button').simulate('click');
    expect(onAddTodoItem.called).toBe(false);
  });

});
