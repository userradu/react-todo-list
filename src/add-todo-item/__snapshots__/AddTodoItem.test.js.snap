// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTodoItem /> renders correctly 1`] = `
<div
  className="add-todo-section"
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
